### PR TITLE
[NativeAOT] Determine IlcHostPackageName from host OS, not target OS

### PR DESCRIPTION
**Pull request by filipnavara**

### CODE REVIEW

File: src/coreclr/nativeaot/BuildIntegration/Microsoft.DotNet.ILCompiler.SingleEntry.targets [hunk 1]
```diff
@@ -15,11 +15,16 @@
...
+    <IlcHostOS Condition="$([MSBuild]::IsOSPlatform('OSX'))">osx</IlcHostOS>
+    <IlcHostOS Condition="$([MSBuild]::IsOSPlatform('FREEBSD'))">freebsd</IlcHostOS>
+    <IlcHostOS Condition="$([MSBuild]::IsOSPlatform('WINDOWS'))">windows</IlcHostOS>
+
     <!-- Default to host that matches OS architecture on Windows. It allows compilation using msbuild.exe that is x86 by default -->
     <IlcHostArch Condition="'$(OS)' == 'Windows_NT' and '$(IlcHostArch)' == ''">$([System.Runtime.InteropServices.RuntimeInformation]::OSArchitecture.ToString().ToLowerInvariant)</IlcHostArch>
     <!-- Default to host that matches SDK architecture on non-Windows -->
     <IlcHostArch Condition="'$(IlcHostArch)' == ''">$([System.Runtime.InteropServices.RuntimeInformation]::ProcessArchitecture.ToString().ToLowerInvariant)</IlcHostArch>
-    <IlcHostPackageName>runtime.$(_targetOSPkg)-$(IlcHostArch).Microsoft.DotNet.ILCompiler</IlcHostPackageName>
+    <IlcHostPackageName>runtime.$(IlcHostOS)-$(IlcHostArch).Microsoft.DotNet.ILCompiler</IlcHostPackageName>
```
Comment (jkotas) [inline src/coreclr/nativeaot/BuildIntegration/Microsoft.DotNet.ILCompiler.SingleEntry.targets:27]: This was the only use of `_targetOSPkg` property. We can delete the `_targetOSPkg` property - revert https://github.com/dotnet/runtime/pull/82590/files.

Comment (filipnavara) [reply src/coreclr/nativeaot/BuildIntegration/Microsoft.DotNet.ILCompiler.SingleEntry.targets:27]: Good catch. I rebased it and didn't notice.

**Comment by jkotas** [comment]
cc @am11

File: src/coreclr/nativeaot/BuildIntegration/Microsoft.DotNet.ILCompiler.SingleEntry.targets (hunk 1 above)
Comment (am11) [reply src/coreclr/nativeaot/BuildIntegration/Microsoft.DotNet.ILCompiler.SingleEntry.targets:27]: Reverting will bring back the linux-musl bug? We don’t have CI coverage for these, so I test it once the next daily build of installer is available.

**Comment by am11** [comment]
This is bringing back the hardcoded list of platforms which we just deleted. Is it fixing a specific scenario?

**Comment by filipnavara** [comment]
> This is bringing back the hardcoded list of platforms which we just deleted. Is it fixing a specific scenario?

It is fixing the scenario of cross-compiling for another OS. For most cases that's moot but it starts to be relevant with the mobile Apple platforms (osx-* host, ios-* target).

**Comment by am11** [comment]
AFAICT, this is not needed in the current `main` branch. We are using `NETCoreSdkPortableRuntimeIdentifier` property to derive `_targetOSPkg` value; which is `osx` for `--framework net8.0-ios`, i.e. the same value what this PR is hard-coding.

**Comment by filipnavara** [comment]
> AFAICT, this is not needed in the current `main` branch. We are using `NETCoreSdkPortableRuntimeIdentifier` property to derive `_targetOSPkg` value; which is `osx` for `--framework net8.0-ios`, i.e. the same value what this PR is hard-coding.

Hmm. Maybe it's correct now afterall. It was not correct when I first made the change. 😅

**Comment by filipnavara** [comment]
I spoke too soon. Now it's broken in completely different way.

`_targetOSPkg` is actually *host*, not target.
`_targetOS` is now set to *host* value too, which breaks the `_IsiOSLikePlatform` checks.

**Comment by jkotas** [comment]
https://github.com/dotnet/runtime/pull/82645 should be fixing this.

**Comment by filipnavara** [comment]
> #82645 should be fixing this.

Yep, I can confirm it fixes the osx-arm64 > maccatalyst-arm64 cross-compilation.

**Pull request closed without merging**